.github/workflows/actions.yml

```diff
@@ -178,3 +178,86 @@ jobs:
           memory: max
       - name: ci/minikube/test-conbench-on-mk.sh
         run: export CONBENCH_REPO_ROOT_DIR=$(pwd) && bash ci/minikube/test-conbench-on-mk.sh
+
+  deploy-packages:
+    runs-on: ubuntu-latest
+    if: github.event_name == 'push'
+    needs:
+      - lint
+      - testsuite
+      - dev-cmds
+      - ui-screenshots
+      - db-migrations
+      - clis
+      - cb-on-minikube
+    matrix:
+      include:
+        - pkg_name: benchadapt
+          pkg_root: ./benchadapt/python
+        - pkg_name: benchclients
+          pkg_root: ./benchclients/python
+        - pkg_name: benchconnect
+          pkg_root: ./benchconnect
+        - pkg_name: benchrun
+          pkg_root: ./benchrun/python
+        - pkg_name: conbench
+          pkg_root: .
+    steps:
+      - name: checkout
+        uses: actions/checkout@v3
+      - name: Set up Python
+        uses: actions/setup-python@v4
+        with:
+          python-version: "3.11"
+      - name: Determine whether version changed
+        id: check
+        run: |
+          last_released_version=$(
+            curl https://pypi.org/pypi/${{ matrix.pkg_name }}/json \
+            | jq '.info.version' \
+            | tr -d '"'
+          )
+
+          cd ${{ matrix.pkg_root }}/${{ matrix.pkg_name }}
+          version=$(
+            python -c 'from _version import __version__; print(__version__)'
+          )
+
+          echo "Last ${{ matrix.pkg_name }} version released to PyPI: $last_released_version"
+          echo "Current ${{ matrix.pkg_name }} version in repo: $version"
+
+          if [[ $last_released_version < $version ]]; then
```

Comment:
Does this comparison work? If yes -- why? :D 

The messy topic of comparing version numbers. Has haunted so many software projects.

Others have built so much code to compare version numbers.
https://stackoverflow.com/a/11887885
https://packaging.pypa.io/en/latest/version.html#packaging.version.parse

Reply:
Ah, I copied this from the former `.buildkite/utils.sh`. That was when they were "sort of" using semver for the legacy Conbench CLI. But the bash `<` just sorts alphabetically. That isn't necessarily what `pip` might do, as you point out.

I think the more intuitive behavior here would just be "check if the version exists on PyPI, and if not, upload it."

Reply:
> I think the more intuitive behavior here would just be "check if the version exists on PyPI, and if not, upload it."

:+1: